moduleTools.sh

```diff
@@ -161,16 +161,24 @@ writeModuleList()
             if [[ -z ${host} ]]; then
                 exitError 654 ${LINENO} "host is not defined"
             fi
-	        # workaround for Todi, Daint, and Lema
-            if [ "${host}" == "lema" -o "${host}" == "todi" -o "${host}" == "daint" ] ; then
-                cat ${modfile} | egrep -v "module load cce\/|module load gcc\/|module load pgi\/" > /tmp/tmp.${host}.${user}.$$
-                compilo=`cat ${modfile} | egrep "module load cce\/|module load gcc\/|module load pgi\/" | sed 's/module load/module swap/g'`
-                echo "${compilo}" >> /tmp/tmp.${host}.${user}.$$
-                /bin/mv -f /tmp/tmp.${host}.${user}.$$ ${modfile}
-	            compilo=`cat ${modfile} | egrep "module load cray-mpich" | sed 's/module load/module swap/g'`
-                sed 's/module load cray-mpich/module swap cray-mpich/g' ${modfile} > /tmp/tmp.${host}.${user}.$$
-	            /bin/mv -f /tmp/tmp.${host}.${user}.$$ ${modfile}
-	        fi
+            # workaround for Todi, Daint, and Lema
+            if [ "${host}" == "lema" -o "${host}" == "daint" ] ; then
+                # Replace some module loads with a swap
+                swap=( "gcc" "pgi" "cce" "cray-mpich" "cudatoolkit" )
+                for i in "${swap[@]}"
+                do
+                    sed -i "s/module load ${i}/module swap ${i}/g" ${modfile}
+                done
+                # Move the swap cudatoolkit to end of file
+                cuda=$(grep cudatoolkit ${modfile})
+                if [ -n cuda ]; then
+                    # Delete the line containing cuda
+                    grep -v "${cuda}" ${modfile} > /tmp/tmp.${host}.${user}.$$
+                    /bin/mv -f /tmp/tmp.${host}.${user}.$$ ${modfile}
+                    # Add it to the end of the file
+                    echo ${cuda} >> ${modfile}
+                fi
+            fi
```

Comment:
actually I am wondering if the machine should not be in a separate if. Do we actually want ot swap gcc on both ?
Otherwise it is ok